### PR TITLE
Add token for admin settings page

### PR DESCRIPTION
**Pull request by jparyani**
This will automatically generate an admin token in the installer and direct users there to configure a login system. Also, added a command `admin-token` to the sandstorm binary to generate said token.

### CODE REVIEW

File: src/sandstorm/util.c++ [hunk 1]
```diff
@@ -601,6 +601,18 @@ kj::Array<byte> base64Decode(kj::StringPtr input) {
 
 // =======================================================================================
 
+kj::String bytesToHex(kj::ArrayPtr<const byte> input) {
```
Comment (kentonv) [inline src/sandstorm/util.c++]: We actually already have hexEncode() in sandstorm-http-bridge.c++. Perhaps move that out to util.{h,c++}. (It's shorter.)

File: src/sandstorm/run-bundle.c++ [hunk 2]
```diff
@@ -992,6 +1003,30 @@ public:
     context.exitInfo(kj::str("reset OAuth configuration"));
   }
 
+  kj::MainBuilder::Validity adminToken() {
+    changeToInstallDir();
+
+    // Get 20 random bytes for token.
+    kj::byte bytes[20];
+    kj::FdInputStream random(raiiOpen("/dev/urandom", O_RDONLY));
```
Comment (kentonv) [inline src/sandstorm/run-bundle.c++]: Use libsodium randombytes_buf(). (We should be switching over to that everywhere.)

File: src/sandstorm/run-bundle.c++ [hunk 3]
```diff
@@ -652,6 +652,17 @@ public:
                   .build();
             },
             "For internal use only.")
+        .addSubCommand("admin-token",
+            [this]() {
+              return kj::MainBuilder(context, VERSION,
+                      "Generates a new admin token that you can use to access the admin settings "
+                      "page. This is meant for initial setup, or if an admin account is locked out.")
+                  .addOption({"short"}, [this]() { shortOutput = true; return true; },
+                      "Pass this flag if you wish to output only the token.")
```
Comment (kentonv) [inline src/sandstorm/run-bundle.c++]: style nitpick: "Pass this flag if you wish to" seems redundant. Just "Output only the token." seems sufficient.

File: src/sandstorm/run-bundle.c++ [hunk 4]
```diff
@@ -652,6 +652,17 @@ public:
                   .build();
             },
             "For internal use only.")
+        .addSubCommand("admin-token",
+            [this]() {
+              return kj::MainBuilder(context, VERSION,
+                      "Generates a new admin token that you can use to access the admin settings "
+                      "page. This is meant for initial setup, or if an admin account is locked out.")
+                  .addOption({"short"}, [this]() { shortOutput = true; return true; },
```
Comment (kentonv) [inline src/sandstorm/run-bundle.c++]: Typically this would be called "quiet", with 'q' as a short option: {'q', "quiet"}

**Comment by kentonv** [comment]
The install script just got rewritten so you'll have to update for that. Sorry.

**Comment by jparyani** [comment]
Done

**Pull request merged**